src/compiler/checker.ts

```diff
@@ -14346,7 +14346,10 @@ namespace ts {
                     let current: Identifier | undefined;
                     while (current = nameStack.shift()) {
                         const meaning = nameStack.length ? SymbolFlags.Namespace : targetMeaning;
-                        const next = getSymbol(getExportsOfSymbol(getMergedSymbol(resolveSymbol(currentNamespace))), current.escapedText, meaning);
+                        // typeof a.b.c is normally resolved using `checkExpression` which in turn defers to `checkQualifiedName`

```

Comment:
the normal resolution procedure is what is *not* happening here, right? I guess that the reason for the change is to make the `typeof` case behave more like the non-`typeof` case. Is that right?

If my guesses are right, then I think it'd be easier to explain this in terms of what actually happens (instead of what normally happens), and why it needs to be similar to the non-`typeof` case.

Reply:
This is to make the `typeof import().a.b` case behave like the `typeof a.b`. Previously, the `typeof import` form essentially did the same thing as the non-typeof form, but for `value` meanings; which seems to _mostly_ work as you'd expect for the `typeof import` case. Until things aren't namespaces, but still have value children, like class statics. :P